qiskit_addon_dice_solver/dice_solver.py

```diff
@@ -46,6 +47,121 @@ def __init__(self, command, returncode, log_path):
+       # OR
+       mpirun_opts = ["-quiet", "-n", "8"]
+
+       energy, sci_coeffs, avg_occs = solve_fermion(..., mpirun_options=mpirun_opts)
+
+    For more information on the ``mpirun`` command line options, refer to the `man page <https://www.open-mpi.org/doc/current/man1/mpirun.1.php>`_.
+
+    .. note::
+
+       Only closed-shell systems are supported. The particle number for both
+       spin-up and spin-down determinants is expected to be equal.
+
+    .. note::
+
+       Determinant are interpreted by the ``Dice`` command line application as 5-byte unsigned integers; therefore, only systems
+       of ``40`` or fewer orbitals are supported.
+
+    Args:
+        bitstring_matrix: A set of configurations defining the subspace onto which the Hamiltonian
+            will be projected and diagonalized. This is a 2D array of ``bool`` representations of bit
+            values such that each row represents a single bitstring. The spin-up configurations
+            should be specified by column indices in range ``(N, N/2]``, and the spin-down
+            configurations should be specified by column indices in range ``(N/2, 0]``, where ``N``
+            is the number of qubits.
+        hcore: Core Hamiltonian matrix representing single-electron integrals
+        eri: Electronic repulsion integrals representing two-electron integrals
+        mpirun_options: Options controlling the CPU resource allocation for the ``Dice`` command line application.
+            These command-line options will be passed directly to the ``mpirun`` command line application during
+            invocation of ``Dice``. These may be formatted as a ``Sequence`` of strings or a single string. If a ``Sequence``,
+            the elements will be combined into a single, space-delimited string and passed to
+            ``mpirun``. If the input is a single string, it will be passed to ``mpirun`` as-is. If no
+            ``mpirun_options`` are provided by the user, ``Dice`` will run on a single MPI slot. For more
+            information on the ``mpirun`` command line options, refer to the `man page <https://www.open-mpi.org/doc/current/man1/mpirun.1.php>`_.
+        working_dir: An absolute path to a directory in which intermediate files can be written to and read from. If
+            no working directory is provided, one will be created using Python's ``tempfile`` module.
+        spin_sq: Target value for the total spin squared for the ground state. If ``None``, no spin will be imposed.
+        clean_working_dir: A flag indicating whether to remove the intermediate files used by the ``Dice``
+            command line application. If ``False``, the intermediate files will be left in a temporary directory in the
+            ``working_dir``.
+
+    Returns:
+        Minimum energy from SCI calculation, SCI coefficients, and average orbital occupancy for spin-up and spin-down orbitals
+    """
+    # Convert bitstring matrix to integer determinants for spin-up/down
```

Comment:
Key deviations from original function:

- Make a working directory in the system's temp directory if one isn't specified
- Create an fci dump from the integrals.
    - SInce we do this, we can ensure there is no nuclear repulsion energy in the fci dump, so we don't have to account for it (same as SQD)
 - Takes a bitstring matrix, rather than ``addresses``. Does not accept ``addresses`` as a deprecated arg.